src/wwwroot/controllers/DomainManagerCtrl.js

```diff
@@ -0,0 +1,46 @@
+(function () {
+    'use strict';
+
+    angular
+      .module('dopplerRelay')
+      .controller('DomainManagerCtrl', DomainManagerCtrl);
+
+    DomainManagerCtrl.$inject = [
+      '$scope',
+      'settings'
+    ];
+
+    function DomainManagerCtrl($scope, settings) {
+      var vm = this;
+      vm.regexDomain = "(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{0,62}[a-zA-Z0-9]\\.)+[a-zA-Z]{2,63}$)";
+      activate();
+
+      function activate() {
+        loadUserDomains();

```

Comment:
Some time ago we have been talking about to try to make `activate` function to return promises, it could be useful in different scenarios, and in general it is not more complex, could you try in this case?